src/order/hom/lattice.lean

```diff
@@ -475,15 +611,60 @@ variables {α}
 
 @[simp] lemma id_apply (a : α) : lattice_hom.id α a = a := rfl
 
+/-- Composition of `lattice_hom`s as a `lattice_hom`. -/
+def comp (f : lattice_hom β γ) (g : lattice_hom α β) : lattice_hom α γ :=
+{ ..f.to_sup_hom.comp g.to_sup_hom, ..f.to_inf_hom.comp g.to_inf_hom }
+
+@[simp] lemma coe_comp (f : lattice_hom β γ) (g : lattice_hom α β) : (f.comp g : α → γ) = f ∘ g :=
+rfl
+@[simp] lemma comp_apply (f : lattice_hom β γ) (g : lattice_hom α β) (a : α) :
+  (f.comp g) a = f (g a) := rfl
+@[simp] lemma coe_comp_sup_hom (f : lattice_hom β γ) (g : lattice_hom α β) :
+  (f.comp g : sup_hom α γ) = (f : sup_hom β γ).comp g := rfl
+@[simp] lemma coe_comp_inf_hom (f : lattice_hom β γ) (g : lattice_hom α β) :
+  (f.comp g : inf_hom α γ) = (f : inf_hom β γ).comp g := rfl
+@[simp] lemma comp_assoc (f : lattice_hom γ δ) (g : lattice_hom β γ) (h : lattice_hom α β) :
+  (f.comp g).comp h = f.comp (g.comp h) := rfl
+@[simp] lemma comp_id (f : lattice_hom α β) : f.comp (lattice_hom.id α) = f :=
+lattice_hom.ext $ λ a, rfl
+@[simp] lemma id_comp (f : lattice_hom α β) : (lattice_hom.id β).comp f = f :=
+lattice_hom.ext $ λ a, rfl
+
+lemma cancel_right {g₁ g₂ : lattice_hom β γ} {f : lattice_hom α β} (hf : function.surjective f) :
+  g₁.comp f = g₂.comp f ↔ g₁ = g₂ :=
+⟨λ h, lattice_hom.ext $ hf.forall.2 $ fun_like.ext_iff.1 h, congr_arg _⟩
+
+lemma cancel_left {g : lattice_hom β γ} {f₁ f₂ : lattice_hom α β} (hg : function.injective g) :
+  g.comp f₁ = g.comp f₂ ↔ f₁ = f₂ :=
+⟨λ h, lattice_hom.ext $ λ a, hg $
+  by rw [←lattice_hom.comp_apply, h, lattice_hom.comp_apply], congr_arg _⟩
+
 end lattice_hom
 
-@[priority 100] -- See note [lower instance priority]
-instance lattice_hom_class.to_inf_hom_class [lattice α] [lattice β] [lattice_hom_class F α β] :
-  inf_hom_class F α β :=
-{ .. ‹lattice_hom_class F α β› }
+namespace order_hom
+variables [linear_order α] [lattice β]
+
+/-- Reinterpret an `order_hom` to a linear order as a `lattice_hom`. -/
+def to_lattice_hom (f : α →o β) : lattice_hom α β :=
```

Comment:
Could you generalize this to a non-instance `order_hom_class.to_lattice_hom_class`, and use the resulting coercion to define this (for all `order_hom_class`es)?

Reply:
I had to include the resulting types in the def because they weren't getting inferred, but I assume that's fine?